common/sagas/wallet.tsx

```diff
@@ -140,32 +142,30 @@ function* unlockMnemonic(action: UnlockMnemonicAction): SagaIterator {
 // inspired by v3:
 // https://github.com/kvhnuke/etherwallet/blob/417115b0ab4dd2033d9108a1a5c00652d38db68d/app/scripts/controllers/decryptWalletCtrl.js#L311
 function* unlockWeb3(): SagaIterator {
-  const failMsg1 = 'Could not connect to MetaMask / Mist.';
-  const failMsg2 = 'No accounts found in MetaMask / Mist.';
-  const { web3 } = window as any;
-
-  if (!web3 || !web3.eth) {
-    yield put(showNotification('danger', translate(failMsg1)));
-    return;
-  }
-
   try {
     yield call(initWeb3Node);
+    yield put(changeNodeIntent('web3'));
+    yield take(
+      action =>
+        action.type === ConfigTypeKeys.CONFIG_NODE_CHANGE &&
+        action.payload.nodeSelection === 'web3'
+    );
 
     const network = NODES.web3.network;
-    const accounts = yield cps(web3.eth.getAccounts);
+    const nodeLib: INode | Web3Node = yield select(getNodeLib);
 
-    if (!accounts.length) {
-      yield put(showNotification('danger', translate(failMsg2)));
-      return;
+    if (!isWeb3Node(nodeLib)) {
+      throw new Error('Cannot use Web3 wallet without a Web3 node.');
     }
 
+    const accounts = yield call([nodeLib, 'getAccounts']);
```

Comment:
Using the string name of the method seems like it would throw off typescript, could you do `call(nodeLib.getAccounts)` instead? If it's a `this` issue, you can do `call(nodeLib.getAccounts.bind(nodeLib))`.

Reply:
Yep, it was a `this` issue. Good point about it throwing off TypeScript, I'll use `bind` instead.

Reply:
Instead you can do `yield apply(nodeLib, nodeLib.getAccounts)`